rootfs/usr/local/bin/kubectl-auto-select

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+function red() {
+	echo "$(tput setaf 1)$*$(tput sgr0)" >&2
+}
+
+function green() {
+	echo "$(tput setaf 2)$*$(tput sgr0)"
+}
+
+# Output the normalized major and minor version number
+# 1. Remove any leading "v"
+# 2. If no dots, assume it is minor version with major version "1"
+function normalize_version() {
+	[[ -n $1 ]] || return
+	if [[ $1 == "null" ]]; then
+		red "Unable to determine Kubernetes (kubectl) version"
+		exit 1
+	fi
+
+	local version=${1#v} # remove leading "v" if present
+	# Customized Kubernetes versions, such as from AWS EKS, may have more than just the
+	# digits in the minor version, e.g. "16+", so strip out everything after leading digits,
+	# including patch version.
+	version=$(sed -E 's/((\d\.)?(\d+))([^d].*)?/\1/' <<<$version)
```

Comment:
I think this function should accept input like `v1.16.whatever`.That way at any time we are dealing with whole "version" string and not just part of it (like minor). It would simplify later on getting version from kubectl:
`kubectl version -o json | jq .serverVersion.gitVersion -j`

If so then I'd use this regex to parse it:
`v?(\d{1}\.\d{2}).*` which seems to be much more clear. 

Or `\d+` to match also future kubectl versions like 1.100 (if it will ever exist).

Reply:
Your version is not that different than mine. The key difference is mine accepts `16` as well as `v1.16.whatever`.